pulp_docker/tests/functional/api/test_recursive_remove.py

```diff
@@ -0,0 +1,319 @@
+# # coding=utf-8
+"""Tests that recursively remove docker content from repositories."""
+import unittest
+
+from pulp_smash import api, config
+from pulp_smash.pulp3.constants import REPO_PATH
+from pulp_smash.pulp3.utils import gen_repo, sync
+from requests.exceptions import HTTPError
+
+from pulp_docker.tests.functional.constants import (
+    DOCKER_TAG_PATH,
+    DOCKER_REMOTE_PATH,
+    DOCKER_RECURSIVE_ADD_PATH,
+    DOCKER_RECURSIVE_REMOVE_PATH,
+    DOCKERHUB_PULP_FIXTURE_1,
+)
+from pulp_docker.tests.functional.utils import gen_docker_remote
+
+
+class TestRecursiveRemove(unittest.TestCase):
+    """Test recursively removing docker content from a repository."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Sync pulp/test-fixture-1 so we can copy content from it."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.from_repo = cls.client.post(REPO_PATH, gen_repo())
+        remote_data = gen_docker_remote(upstream_name=DOCKERHUB_PULP_FIXTURE_1)
+        cls.remote = cls.client.post(DOCKER_REMOTE_PATH, remote_data)
+        sync(cls.cfg, cls.remote, cls.from_repo)
+        latest_version = cls.client.get(cls.from_repo['_href'])['_latest_version_href']
+        cls.latest_from_version = "repository_version={version}".format(version=latest_version)
+
+    def setUp(self):
+        """Create an empty repository to copy into."""
+        self.to_repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, self.to_repo['_href'])
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete things made in setUpClass. addCleanup feature does not work with setupClass."""
```

Comment:
FYI. Pulp Smash has a function `ensure_teardownclass` that will assure tearDownClass will be executes, even when an exception happens during the `setUpClass`.

Reply:
Should I replace some of the setUpClass with a context manager like this?
https://github.com/pulp/pulpcore/blob/ae6be0d89c5df26ac1cfdc876f3b131aa6e9bcf8/pulpcore/tests/functional/api/test_pagination.py#L38

Reply:
IRC discussion: something to know about, but not necessary in this PR